Index: sys/dev/atkbdc/atkbd.c
===================================================================
--- sys/dev/atkbdc/atkbd.c	(revision 222110)
+++ sys/dev/atkbdc/atkbd.c	(working copy)
@@ -44,7 +44,7 @@
 #include <machine/bus.h>
 #include <machine/resource.h>
 
-#if defined(__i386__) || defined(__amd64__)
+#if defined(__i386__)
 #include <machine/md_var.h>
 #include <machine/psl.h>
 #include <compat/x86bios/x86bios.h>
@@ -55,7 +55,7 @@
 #include <vm/vm_param.h>
 
 #include <isa/isareg.h>
-#endif /* __i386__ || __amd64__ */
+#endif /* __i386__ */
 
 #include <sys/kbio.h>
 #include <dev/kbd/kbdreg.h>
@@ -1089,7 +1089,7 @@
 static int
 get_typematic(keyboard_t *kbd)
 {
-#if defined(__i386__) || defined(__amd64__)
+#if defined(__i386__)
 	/*
 	 * Only some systems allow us to retrieve the keyboard repeat
 	 * rate previously set via the BIOS...
@@ -1127,7 +1127,7 @@
 	return (0);
 #else
 	return (ENODEV);
-#endif /* __i386__ || __amd64__ */
+#endif /* __i386__ */
 }
 
 static int
